### PR TITLE
Windows 2015 warnings rollup

### PR DESCRIPTION
**Pull request by hobu**
Related to #1361, I got a MSVC 2015 build system going and attempted to clean up all of the warnings it was complaining about so we have a better chance of seeing changes in the output as development continues.

### CODE REVIEW

File: src/SpatialReference.cpp [hunk 1]
```diff
@@ -362,7 +362,7 @@ int SpatialReference::calculateZone(double lon, double lat)
     // Everywhere else.
     else
     {
-        zone = floor((lon + 180.0) / 6) + 1;
+        zone = (int) floor((lon + 180.0) / 6) + 1;
```
Comment (abellgithub) [inline src/SpatialReference.cpp]: It complains about THIS?  So silly.  Makes me shudder for thoughts of lint.

File: filters/transformation/TransformationFilter.cpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ std::string TransformationFilter::getName() const { return s_info.name; }
 TransformationMatrix transformationMatrixFromString(const std::string& s)
 {
     std::istringstream iss(s);
-    TransformationMatrix matrix;
+	TransformationMatrix matrix{ 0 };
```
Comment (abellgithub) [inline filters/transformation/TransformationFilter.cpp]: Sometimes the programmer does know better :)

File: include/pdal/PDALUtils.hpp [hunk 3]
```diff
@@ -106,7 +106,7 @@ inline Everything extractDim(Extractor& ext, Dimension::Type type)
 {
     using Type = Dimension::Type;
 
-    Everything e;
+	Everything e{ 0 };
```
Comment (abellgithub) [inline include/pdal/PDALUtils.hpp]: This is a bit of performance sensitive code.  The optimizer may eliminate the unnecessary initialization, but I'm not sure.  Certainly this is an unnecessary initialization, though.

Comment (abellgithub) [inline include/pdal/PDALUtils.hpp]: Can we just suppress the warning?

File: filters/transformation/TransformationFilter.cpp (hunk 2 above)
Comment (hobu) [inline filters/transformation/TransformationFilter.cpp]: backed out in @ac3e4fb

**Pull request merged**